src/errors/ResponseError.js

```diff
@@ -1,20 +1,75 @@
 // @flow
 
-class ResponseError extends Error {
-  response: Response
+type ResponseErrorParamsType = {|
+  endpointName: string,
+  response: Response,
+  url: string,
+  formData: ?FormData
+|}
 
-  getMessage = (endpointName: string, response: Response): string =>
-    `ResponseError: Failed to load the request for the ${endpointName} endpoint. Response status ${response.status}`
+class ResponseError extends Error {
+  _endpointName: string
+  _response: Response
+  _url: string
+  _formData: ?FormData
+  _message: string
 
-  constructor (params: { endpointName: string, response: Response }) {
+  constructor (params: ResponseErrorParamsType) {
     super()
 
     if (Error.captureStackTrace) {
       Error.captureStackTrace(this, ResponseError)
     }
 
-    this.response = params.response
-    this.message = this.getMessage(params.endpointName, params.response)
+    const { endpointName, response, url, formData } = params
+
+    this._message = this.createMessage(params)
+    this._endpointName = endpointName
+    this._response = response
+    this._url = url
+    this._formData = formData
+  }
+
+  createMessage ({ formData, url, endpointName, response }: ResponseErrorParamsType): string {
+    const stringifyFormData = (formData: ?FormData) => {
+      if (formData) {
+        const entries = {}
+        for (const entry of formData.entries()) {
+          entries[entry[0]] = entry[1]
```

Comment:
do we ever fetch 'multipart/form-data' (files), because in this case it would be good to liit the lenght of the entries.

Reply:
Not that I know of, no. Actually, form-data is just used to post feedback.